src/shared/services/bolt/bolt.ts

```diff
@@ -82,135 +77,104 @@ function routedWriteTransaction(
-      cancelable,
+  const id = requestId || v4()
+  const payload = getWorkerPayloadForRunningCypherMessage(
+    input,
+    mappings.recursivelyTypeGraphItems(parameters),
+    boltConnection.ROUTED_WRITE_CONNECTION,
+    id,
+    cancelable,
+    {
+      ...connectionProperties,
       txMetadata,
       useDb: useDb || _useDb,
       autoCommit
-    })
-  }
+    }
+  )
+  const workerPromise = setupBoltWorker(
+    boltWorkPool,
+    id,
+    payload,
+    onLostConnection
+  )
+  return [id, workerPromise]
 }
 
 function routedReadTransaction(
   input: any,
   parameters: any,
   requestMetaData: any = {}
-): any {
+): Promise<QueryResult> {
   const {
-    useCypherThread = false,
     requestId = null,
     cancelable = false,
     onLostConnection = () => {},
     txMetadata = undefined,
     useDb = null
   } = requestMetaData
-  if (useCypherThread && window.Worker) {
-    const id = requestId || v4()
-    const payload = getWorkerPayloadForRunningCypherMessage(
-      input,
-      mappings.recursivelyTypeGraphItems(parameters),
-      boltConnection.ROUTED_READ_CONNECTION,
-      id,
-      cancelable,
-      {
-        ...connectionProperties,
-        txMetadata,
-        useDb: useDb || _useDb
-      }
-    )
-    const workerPromise = setupBoltWorker(
-      boltWorkPool,
-      id,
-      payload,
-      onLostConnection
-    )
-    return workerPromise
-  } else {
-    return globalRoutedReadTransaction(input, parameters, {
-      requestId,
-      cancelable,
+  const id = requestId || v4()
+  const payload = getWorkerPayloadForRunningCypherMessage(
+    input,
+    mappings.recursivelyTypeGraphItems(parameters),
+    boltConnection.ROUTED_READ_CONNECTION,
+    id,
+    cancelable,
+    {
+      ...connectionProperties,
       txMetadata,
       useDb: useDb || _useDb
-    })
-  }
+    }
+  )
+  const workerPromise = setupBoltWorker(
+    boltWorkPool,
+    id,
+    payload,
+    onLostConnection
+  )
+  return workerPromise
 }
 
 function directTransaction(
   input: any,
   parameters: any,
   requestMetaData: any = {}
-): any {
+): Promise<QueryResult> {
   const {
-    useCypherThread = false,
     requestId = null,
     cancelable = false,
     onLostConnection = () => {},
     txMetadata = undefined,
     useDb = null
   } = requestMetaData
-  if (useCypherThread && window.Worker) {
-    const id = requestId || v4()
-    const payload = getWorkerPayloadForRunningCypherMessage(
-      input,
-      mappings.recursivelyTypeGraphItems(parameters),
-      boltConnection.DIRECT_CONNECTION,
-      id,
-      cancelable,
-      {
-        ...connectionProperties,
-        txMetadata,
-        useDb: useDb || _useDb
-      }
-    )
-    const workerPromise = setupBoltWorker(
-      boltWorkPool,
-      id,
-      payload,
-      onLostConnection
-    )
-    return workerPromise
-  } else {
-    return globalDirectTransaction(input, parameters, {
-      requestId,
-      cancelable,
+  const id = requestId || v4()
```

Comment:
should we handle the case where the user is some really old browser that doesn't have window.Worker?

Reply:
I could throw an error I guess. Otherwise I'm not sure. Do we have any fallback currently for unsupported browsers?

Reply:
We don't have any other fallbacks, but what we could do is warn earlier in the application. I think it'd make sense to change `:play start` to a message warning about the browser being unsupported or something like that

Reply:
It'd say it's not something you'd need to do